### PR TITLE
Add a diagnostics job to the CodeQL workflow

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description ##

This pull request adds a diagnostics job to the CodeQL workflow.

## 💭 Motivation and context ##

The CodeQL workflow already had a harden-runner task, but it's good to agree everywhere with the changes we made to the build.yml workflow in cisagov/skeleton-generic#144.

## 🧪 Testing ##

All automated tests pass.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

### CODE REVIEW

**Review by dv4harr10 — APPROVED** [review]

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -37,12 +56,6 @@ jobs:
         # https://docs.github.com/en/github/finding-security-vulnerabilities-and-errors-in-your-code/configuring-code-scanning#overriding-automatic-language-detection
 
     steps:
-      - id: harden-runner
```
Comment (mcdonnnj) [inline .github/workflows/codeql-analysis.yml]: I do not believe this should be removed.

**Review by mcdonnnj — CHANGES_REQUESTED** [review]

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (jsf9k) [reply .github/workflows/codeql-analysis.yml]: Doh!  You are correct.  Fixed in commit 2e30384.

**Review by dv4harr10 — APPROVED** [review]

**Review by mcdonnnj — APPROVED** [review]
LGTM ✔

**Review by dav3r — APPROVED** [review]
🚦

**Pull request merged**